app/script/connect/ConnectRepository.coffee

```diff
@@ -121,33 +120,33 @@ class z.connect.ConnectRepository
   @return [z.connect.PhoneBook] Encoded phone book data
   ###
   _parse_macos_contacts: ->
-    return if not window.zAddressBook
-
-    address_book = window.zAddressBook()
-    phone_book = new z.connect.PhoneBook @properties_repository.self()
+    return new Promise (resolve, reject) =>
+      return resolve undefined if not window.zAddressBook
+      address_book = window.zAddressBook
```

Comment:
@lipis old mac clients will still expose zAddressbook as a Function, so you should take that into account by checking for the new and the old syntax.